Source/WebCore/page/SecurityOrigin.cpp

```diff
@@ -179,18 +160,48 @@ SecurityOrigin::SecurityOrigin(const SecurityOrigin* other)
 
 Ref<SecurityOrigin> SecurityOrigin::create(const URL& url)
 {
-    if (RefPtr<SecurityOrigin> cachedOrigin = getCachedOrigin(url))
-        return cachedOrigin.releaseNonNull();
+    // FIXME: We ought to be able to assert that isInWebProcess() is true, but
+    // WebKitLegacy doesn't have a web content process.
+    // This is for LegacySchemeRegistry::schemeIsHandledBySchemeHandler.
```

Comment:
Something like this seems guaranteed to get out of sync. I wish there was a way to out the assertions right at the code that uses the global state but also have the benefit of checking when we take other paths. I guess we can leave like this for now.

Reply:
Unfortunately this doesn't seem to work at all. See all the failures on the debug bot. Not sure how to proceed with this.